### PR TITLE
feat: bump sdk 3.6.7 and switch to gomod

### LINKED ISSUE #60: [Repolinter] Open Source Policy Issues
# Repolinter Report

🤖*This issue was automatically generated by [repolinter-action](https://github.com/newrelic/repolinter-action), developed by the Open Source and Developer Advocacy team at New Relic. This issue will be automatically updated or closed when changes are pushed. If you have any problems with this tool, please feel free to open a GitHub issue or give us a ping in #help-opensource.*

This Repolinter run generated the following results:
| ❗  Error | ❌  Fail | ⚠️  Warn | ✅  Pass | Ignored | Total |
|---|---|---|---|---|---|
| 0 | 1 | 0 | 6 | 0 | 7 |

- [Fail](#user-content-fail)
  - [❌ `readme-starts-with-community-plus-header`](#user-content--readme-starts-with-community-plus-header)
- [Passed](#user-content-passed)
  - [✅ `license-file-exists`](#user-content--license-file-exists)
  - [✅ `readme-file-exists`](#user-content--readme-file-exists)
  - [✅ `readme-contains-link-to-security-policy`](#user-content--readme-contains-link-to-security-policy)
  - [✅ `readme-contains-discuss-topic`](#user-content--readme-contains-discuss-topic)
  - [✅ `code-of-conduct-should-not-exist-here`](#user-content--code-of-conduct-should-not-exist-here)
  - [✅ `third-party-notices-file-exists`](#user-content--third-party-notices-file-exists)

### ❌ `readme-starts-with-community-plus-header` <a href="#user-content--readme-starts-with-community-plus-header" id="-readme-starts-with-community-plus-header">#</a>

The README of a community plus project should have a community plus header at the start of the README. If you already have a community plus header and this rule is failing, your header may be out of date, and you should update your header with the suggested one below. For more information please visit https://opensource.newrelic.com/oss-category/. Below is a list of files or patterns that failed:

- `README.md`: The first 5 lines do not contain the pattern(s): Open source Community Plus header (see https://opensource.newrelic.com/oss-category).
  - 🔨 **Suggested Fix**: prepend `[Community Plus header](https://opensource.newrelic.com/oss-category/#community-plus)` to file

## Passed <a href="#user-content-passed" id="passed">#</a>

<details>
<summary>Click to see rules</summary>

### ✅ `license-file-exists` <a href="#user-content--license-file-exists" id="-license-file-exists">#</a>

Found file (`LICENSE`). New Relic requires that all open source projects have an associated license contained within the project. This license must be permissive (e.g. non-viral or copyleft), and we recommend Apache 2.0 for most use cases. For more information please visit https://docs.google.com/document/d/1vML4aY_czsY0URu2yiP3xLAKYufNrKsc7o4kjuegpDw/edit.

### ✅ `readme-file-exists` <a href="#user-content--readme-file-exists" id="-readme-file-exists">#</a>

Found file (`README.md`). New Relic requires a README file in all projects. This README should give a general overview of the project, and should point to additional resources (security, contributing, etc.) where developers and users can learn further. For more information please visit https://github.com/newrelic/open-by-default.

### ✅ `readme-contains-link-to-security-policy` <a href="#user-content--readme-contains-link-to-security-policy" id="-readme-contains-link-to-security-policy">#</a>

Contains a link to the security policy for this repository (`README.md`). New Relic recommends putting a link to the open source security policy for your project (`https://github.com/newrelic/<repo-name>/security/policy` or `../../security/policy`) in the README. For an example of this, please see the "a note about vulnerabilities" section of the [Open By Default repository](https://github.com/newrelic/open-by-default#contribute). For more information please visit https://nerdlife.datanerd.us/new-relic/security-guidelines-for-publishing-source-code.

### ✅ `readme-contains-discuss-topic` <a href="#user-content--readme-contains-discuss-topic" id="-readme-contains-discuss-topic">#</a>

Contains a link to the appropriate discuss.newrelic.com topic (`README.md`). New Relic recommends directly linking the your appropriate discuss.newrelic.com topic in the README, allowing developer an alternate method of getting support. For more information please visit https://nerdlife.datanerd.us/new-relic/security-guidelines-for-publishing-source-code.

### ✅ `code-of-conduct-should-not-exist-here` <a href="#user-content--code-of-conduct-should-not-exist-here" id="-code-of-conduct-should-not-exist-here">#</a>

New Relic has moved the `CODE_OF_CONDUCT` file to a [centralized location](https://github.com/newrelic/.github/blob/main/CODE_OF_CONDUCT.md) where it is referenced automatically by every repository in the New Relic organization. Because of this change, any other `CODE_OF_CONDUCT` file in a repository is now redundant and should be removed. Note that you will need to adjust any links to the local `CODE_OF_CONDUCT` file in your documentation to point to the central file (`README` and `CONTRIBUTING` will probably have links that need updating). For more information please visit https://docs.google.com/document/d/1y644Pwi82kasNP5VPVjDV8rsmkBKclQVHFkz8pwRUtE/view. Did not find a file matching the specified patterns. All files passed this test.

### ✅ `third-party-notices-file-exists` <a href="#user-content--third-party-notices-file-exists" id="-third-party-notices-file-exists">#</a>

Found file (`THIRD_PARTY_NOTICES.md`). A [`THIRD_PARTY_NOTICES.md`](https://github.com/newrelic/opensource-website/blob/develop/THIRD_PARTY_NOTICES.md) file can be present in your repository to grant attribution to all dependencies being used by this project. This document is necessary if you are using third-party source code in your project, with the exception of code referenced outside the project's compiled/bundled binary (ex. some Java projects require modules to be pre-installed in the classpath, outside the project binary and therefore outside the scope of the `THIRD_PARTY_NOTICES`). Please review your project's dependencies and create a THIRD_PARTY_NOTICES.md file if necessary. For JavaScript projects, you can generate this file using the [oss-cli](https://github.com/newrelic/newrelic-oss-cli). For more information please visit https://docs.google.com/document/d/1y644Pwi82kasNP5VPVjDV8rsmkBKclQVHFkz8pwRUtE/view.

</details>

### LINKED ISSUE #61: bump integration sdk and switch to go modules

### PR DESCRIPTION
**Pull request by gsanchezgavier**
- [x] - removing obsolete win-config file
- [x] - bump to go 1.16
- [x] - use go mod
- [x] - bump dependencies
- [x] - Adds json schema integration tests
- [x] - fix cluster_name collision adding mongodb_cluster_name (same [workaround](https://github.com/newrelic/nri-haproxy/pull/29/files) as nri-haproxy)
- [ ] - Add golanci-lint in ci/validate (de-scoped)
Fixes #61 and Fixes #60

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,52 +19,51 @@ clean:
...
-
-deps-only:
-	@echo "=== $(INTEGRATION) === [ deps ]: Installing package dependencies required by the project..."
-	@govendor sync
+deps: tools 
 
 validate: deps
 	@echo "=== $(INTEGRATION) === [ validate ]: Validating source code running gometalinter..."
-	@gometalinter.v2 --config=.gometalinter.json $(GO_FILES)...
+
```
Comment (roobre) [inline Makefile:31]: Could we garbage-collect empty targets like this? Might be confusing for us to see later otherwise.

File: build/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
-FROM golang:1.15.2-buster
+FROM golang:1.16-buster
 
-ARG GH_VERSION='1.4.0'
+ARG GH_VERSION='1.6.0'
```
Comment (roobre) [inline build/Dockerfile:3]:
```suggestion
ARG GH_VERSION='1.9.2'
```

**Comment by roobre** [comment]
Looking good other than the two small comments above!

File: Makefile (hunk 1 above)
Comment (gsanchezgavier) [reply Makefile:31]: oh this was just for a test and forgot to add them again, i descoped the linter on the PR

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,7 +4,14 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
-
+## 2.6.0 (2021-04-26)
+### Changed
+- Decorate metrics with `mongodbClusterName`
+  `clusterName` (deprecated) attribute might get overwritten by the Kubernetes integration when running inside kubernetes.
```
Comment (davidbrota) [inline CHANGELOG.md:10]: isn't the agent that overwrites the clusterName?

**Review by davidbrota — COMMENTED** [review]
LGTM, but I'm getting this error when executing make
```
=== mongodb === [ validate ]: Validating source code running gometalinter...
WARNING: deadline exceeded by linter goimports (try increasing --deadline)
make: *** [validate] Error 2
```

File: CHANGELOG.md (hunk 3 above)
Comment (gsanchezgavier) [reply CHANGELOG.md:10]: yes is true, i'll change that thanks.

**Comment by gsanchezgavier** [comment]
@davidbrota  I extended a timeout on the linter that fixed the issue. Anyway the linter will be replaced soon.

**Review by davidbrota — APPROVED** [review]

**Comment by pyrotechnics-io** [comment]
The docker container doesnt seem to build on Go 1.16 any more because of deprecation errors on x509.DecryptPEMBlock which prevent the build from going further. 

```

#9 46.85 level=info msg="[loader] Go packages loading at mode 575 (compiled_files|imports|deps|exports_file|files|name|types_sizes) took 2.429960877s"
#9 46.86 level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 7.227689ms"
#9 50.62 level=info msg="[linters context/goanalysis] analyzers took 9.522637668s with top 10 stages: buildir: 6.960595062s, inspect: 714.961085ms, fact_deprecated: 325.439419ms, nilness: 304.690269ms, fact_purity: 232.113214ms, ctrlflow: 225.9055ms, printf: 191.95837ms, SA5012: 157.362513ms, typedness: 114.898051ms, unused: 25.619011ms"
#9 50.62 level=info msg="[runner] Processors filtering stat (out/in): path_shortener: 2/2, skip_files: 2/2, exclude-rules: 2/2, diff: 2/2, max_from_linter: 2/2, uniq_by_line: 2/2, source_code: 2/2, severity-rules: 2/2, cgo: 2/2, path_prettifier: 2/2, identifier_marker: 2/2, nolint: 2/2, path_prefixer: 2/2, sort_results: 2/2, filename_unadjuster: 2/2, exclude: 2/2, max_per_file_from_linter: 2/2, max_same_issues: 2/2, skip_dirs: 2/2, autogenerated_exclude: 2/2"
#9 50.62 level=info msg="[runner] processing took 1.015144ms with stages: exclude-rules: 431.446µs, nolint: 308.239µs, identifier_marker: 138.884µs, path_prettifier: 28.891µs, autogenerated_exclude: 27.291µs, skip_dirs: 25.937µs, uniq_by_line: 23.296µs, source_code: 21.543µs, max_same_issues: 2.535µs, cgo: 1.625µs, path_shortener: 1.168µs, filename_unadjuster: 1.013µs, max_from_linter: 955ns, diff: 681ns, max_per_file_from_linter: 462ns, skip_files: 382ns, severity-rules: 260ns, exclude: 236ns, sort_results: 175ns, path_prefixer: 125ns"
#9 50.62 level=info msg="[runner] linters took 3.757774376s with stages: goanalysis_metalinter: 3.756611493s"
#9 50.62 level=info msg="File cache stats: 1 entries of total size 4.9KiB"
#9 50.62 src/connection/info.go:187:5: SA1019: x509.IsEncryptedPEMBlock has been deprecated since Go 1.16 because it shouldn't be used: Legacy PEM encryption as specified in RFC 1423 is insecure by design. Since it does not authenticate the ciphertext, it is vulnerable to padding oracle attacks that can let an attacker recover the plaintext. (staticcheck)
#9 50.62        if x509.IsEncryptedPEMBlock(pvtKeyBlock) {
#9 50.62           ^
#9 50.62 src/connection/info.go:188:23: SA1019: x509.DecryptPEMBlock has been deprecated since Go 1.16 because it shouldn't be used: Legacy PEM encryption as specified in RFC 1423 is insecure by design. Since it does not authenticate the ciphertext, it is vulnerable to padding oracle attacks that can let an attacker recover the plaintext. (staticcheck)
#9 50.62                decPemBytes, err := x509.DecryptPEMBlock(pvtKeyBlock, []byte(passphrase))
#9 50.62                                    ^
#9 50.62 level=info msg="Memory: 64 samples, avg is 171.2MB, max is 341.6MB"
#9 50.62 level=info msg="Execution took 6.204165876s"
#9 50.63 exit status 1
#9 50.71 make: *** [Makefile:22: validate] Error 1
------
executor failed running [/bin/sh -c cd /go/src/github.com/newrelic/nri-mongodb &&     make &&     strip ./bin/nri-mongodb]: exit code: 2
The terminal process "/usr/bin/bash '-c', 'docker build --pull --rm -f "Dockerfile" -t nrimongodb:latest "."'" terminated with exit code: 1.

```

**Comment by gsanchezgavier** [comment]
@pyrotechnics-io  we are currently bumping some dependencies on this repo and removing this Dockerfile since this is not something that is being recommended to use. 
There is an image [infrastructure-bundle](https://github.com/newrelic/infrastructure-bundle) hosted on [dockerhub](https://hub.docker.com/r/newrelic/infrastructure-bundle) that contains the Infrastructure Agent plus all supported newrelic integrations.

**Comment by pyrotechnics-io** [comment]
Ah alright. Thank you very much for the info. I'll do that.

**Pull request merged**